src/auto-evo/simulation/SimulationCache.cs

```diff
@@ -95,148 +99,214 @@ public float GetBaseHexSizeForSpecies(MicrobeSpecies species)
-                        * Math.Pow(absoluteOutput, Constants.AUTO_EVO_ABSOLUTE_PRODUCTION_POWER_BIAS)
-                        * Constants.AUTO_EVO_ATP_USE_SCORE_MULTIPLIER);
-                }
+        var key = (microbeSpecies, prey, biomeConditions);
+
+        if (predationScores.TryGetValue(key, out var cached))
+        {
+            return cached;
+        }
+
+        var preyHexSize = GetBaseHexSizeForSpecies(prey);
+        var preySpeed = GetBaseSpeedForSpecies(prey);
+
+        var behaviourScore = microbeSpecies.Behaviour.Aggression / Constants.MAX_SPECIES_AGGRESSION;
+
+        // TODO: if these two methods were combined it might result in better performance with needing just
+        // one dictionary lookup
+        var microbeSpeciesHexSize = GetBaseHexSizeForSpecies(microbeSpecies);
+        var predatorSpeed = GetBaseSpeedForSpecies(microbeSpecies);
+
+        // Only assign engulf score if one can actually engulf
+        var engulfScore = 0.0f;
+        if (microbeSpeciesHexSize / preyHexSize >
+            Constants.ENGULF_SIZE_RATIO_REQ && microbeSpecies.CanEngulf)
+        {
+            // Catch scores grossly accounts for how many preys you catch in a run;
+            var catchScore = 0.0f;
+
+            // First, you may hunt individual preys, but only if you are fast enough...
+            if (predatorSpeed > preySpeed)
+            {
+                // You catch more preys if you are fast, and if they are slow.
+                // This incentivizes engulfment strategies in these cases.
+                catchScore += predatorSpeed / preySpeed;
             }
+
+            // ... but you may also catch them by luck (e.g. when they run into you),
+            // and this is especially easy if you're huge.
+            // This is also used to incentivize size in microbe species.
+            catchScore += Constants.AUTO_EVO_ENGULF_LUCKY_CATCH_PROBABILITY * microbeSpeciesHexSize;
+
+            // Allow for some degree of lucky engulfment
+            engulfScore = catchScore * Constants.AUTO_EVO_ENGULF_PREDATION_SCORE;
         }
 
-        cachedEnergyCreationScoreForOrganelle.Add(key, energyCreationScore);
-        return energyCreationScore;
-    }
+        var (pilusScore, oxytoxyScore, mucilageScore) = GetPredationToolsRawScores(microbeSpecies);
 
-    /// <summary>
-    ///   A measure of how good the species is for generating energy from a given compound.
-    /// </summary>
-    /// <returns>
-    ///   A float to represent score. Scores are only compared against other scores from the same FoodSource,
-    ///   so different implementations do not need to worry about scale.
-    /// </returns>
-    public float GetEnergyGenerationScoreForSpecies(MicrobeSpecies species, BiomeConditions biomeConditions,
-        Compound compound)
-    {
-        var key = (species, biomeConditions, compound);
+        // don't use mucus for now
```

Comment:
Maybe a TODO comment here? Also I noticed there's a few comments in around this line that don't start with a capital letter.